src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitSchedulerSessionsService.kt

```diff
@@ -0,0 +1,164 @@
+              availableSession.sessionTimeSlot.endTime,
+              overridingAppointments.size,
+              getOverridingEventDetails(overridingAppointments),
+            )
+          }
+        }
+      }
+    } else {
+      availableVisitSessions
+    }
+
+    return availableVisitSessions.sortedWith(
+      compareBy(
+        { it.sessionDate },
+        { it.sessionTimeSlot.startTime },
+        { it.sessionTimeSlot.endTime },
+      ),
+    )
+  }
+
+  fun getSessionCapacity(
+    prisonCode: String,
+    sessionDate: LocalDate,
+    sessionStartTime: LocalTime,
+    sessionEndTime: LocalTime,
+  ): SessionCapacityDto? {
+    return visitSchedulerClient.getSessionCapacity(prisonCode, sessionDate, sessionStartTime, sessionEndTime)
+  }
+
+  fun getSessionSchedule(prisonCode: String, sessionDate: LocalDate): List<SessionScheduleDto>? {
+    return visitSchedulerClient.getSessionSchedule(prisonCode, sessionDate)
+  }
+
+  private fun updateRequestedRestriction(
+    requestedSessionRestriction: SessionRestriction?,
+    prisonerId: String,
+    visitors: List<Long>?,
+  ): SessionRestriction {
+    return if (prisonerProfileService.hasPrisonerGotClosedRestrictions(prisonerId) ||
+      (if (visitors != null) prisonerProfileService.hasVisitorsGotClosedRestrictions(prisonerId, visitors) else false)
+    ) {
+      CLOSED
+    } else {
+      requestedSessionRestriction ?: OPEN
+    }
+  }
+
+  private fun getHigherPriorityAppointmentsForDate(
+    higherPriorityAppointments: List<ScheduledEventDto>,
+    sessionDate: LocalDate,
+  ): List<ScheduledEventDto> {
+    return higherPriorityAppointments.filter { it.eventDate == sessionDate }
+  }
+
+  private fun getOverridingAppointments(
+    visitSession: AvailableVisitSessionDto,
+    higherPriorityEvents: List<ScheduledEventDto>,
+  ): List<ScheduledEventDto> {
+    return higherPriorityEvents.filter { hasOverridingAppointment(visitSession, it) }
+  }
+
+  private fun hasOverridingAppointment(
+    visitSession: AvailableVisitSessionDto,
+    scheduledEvent: ScheduledEventDto,
+  ): Boolean {
+    val visitDate = visitSession.sessionDate
+    val visitStartTime = visitSession.sessionTimeSlot.startTime
+    val visitEndTime = visitSession.sessionTimeSlot.endTime
+
+    return (
+      (scheduledEvent.eventDate != null && visitDate == scheduledEvent.eventDate) &&
+        hasOverlappingTimes(
+          scheduledEvent.startTime?.toLocalTime() ?: LocalTime.MIN,
+          scheduledEvent.endTime?.toLocalTime() ?: LocalTime.MAX,
+          visitStartTime,
+          visitEndTime,
+        )
+      )
+  }
+
+  private fun getOverridingEventDetails(events: List<ScheduledEventDto>): String {
+    return events.joinToString(" and ")
+  }
+
+  private fun hasOverlappingTimes(
+    scheduledEventStartTime: LocalTime,
+    scheduledEventEndTime: LocalTime,
+    visitSessionStartTime: LocalTime,
+    visitSessionEndTime: LocalTime,
+  ): Boolean {
+    return (
```

Comment:
return (
    visitSessionStartTime < scheduledEventEndTime &&
    visitSessionEndTime > scheduledEventStartTime
);

Could this be simplified to these conditions? As there is a bit of overlap with the 3 conditionals. If we have these 2 checks, it should cover it all. Double check with the existing unit tests

Reply:
also might be possible to make it more readable by doing

return (
visitSessionStartTime.isBefore(scheduledEventEndTime) && visitSessionEndTime.isAfter(scheduledEventStartTime)
);

Reply:
nice. the tests work. need to get my head around how it works first :)

Reply:
Updated. Thanks!